File: app/src/main/java/com/recursia/popularmovies/presentation/views/contracts/MainScreenContract.java
package com.recursia.popularmovies.presentation.views.contracts;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface MainScreenContract extends MvpView {

    void openAboutDialog();

    void hideAboutDialog();

    @StateStrategyType(SkipStrategy.class)
    void openGooglePlayPage();

}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/contracts/MainScreenContract.kt
package com.recursia.popularmovies.presentation.views.contracts

import com.recursia.popularmovies.domain.models.Movie
import com.recursia.popularmovies.domain.models.enums.Category
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndStrategy
import moxy.viewstate.strategy.SkipStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndStrategy::class)
interface MainScreenContract : MvpView {
    fun setCategoryMovies(movies: List<Movie>, category: Category)

    @StateStrategyType(AddToEndStrategy::class)
    fun addCategoryMovies(movies: List<Movie>, category: Category)

    @StateStrategyType(SkipStrategy::class)
    fun showErrorMessage(message: String)
}
